src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../features/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+type Props = {
+  currentTodo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ currentTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatch = useAppDispatch();
+
+  const deleteCurrentTodo = () =>
+    dispatch(currentTodoActions.deleteCurrentTodo());
```

Comment:
The `deleteCurrentTodo` function is used to close the modal by dispatching an action. Ensure that this action correctly updates the state to close the modal. If the intention is to remove the current todo, verify that this action is implemented correctly in the Redux slice.